springwolf-core/src/test/java/io/github/stavshamir/springwolf/schemas/example/DefaultSchemaWalkerXmlIntegrationTest.java

```diff
@@ -0,0 +1,525 @@
+        void type_string_format_password() {
+            PasswordSchema schema = new PasswordSchema();
+            schema.setName("type_string_format_password");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual).isEqualTo("<type_string_format_password>string-password</type_string_format_password>");
+        }
+
+        @Test
+        void type_string_format_uuid() {
+            UUIDSchema schema = new UUIDSchema();
+            schema.setName("type_string_format_password");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual)
+                    .isEqualTo(
+                            "<type_string_format_password>3fa85f64-5717-4562-b3fc-2c963f66afa6</type_string_format_password>");
+        }
+
+        @Test
+        void type_string_format_unknown() {
+            StringSchema schema = new StringSchema();
+            schema.setFormat("unknown");
+            schema.setName("type_string_format_unknown");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual)
+                    .isEqualTo(
+                            "<type_string_format_unknown>unknown string schema format: unknown</type_string_format_unknown>");
+        }
+
+        @Test
+        void type_unknown_schema() {
+            class TestSchema extends Schema<StringSchema> {
+                TestSchema() {
+                    super("test-schema", (String) null);
+                }
+            }
+
+            TestSchema schema = new TestSchema();
+            schema.setName("type_unknown_schema");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual).isEqualTo("<type_unknown_schema>unknown schema type: test-schema</type_unknown_schema>");
+        }
+
+        @Test
+        void type_primitive_array() {
+            ArraySchema schema = new ArraySchema();
+            schema.setItems(new StringSchema());
+            schema.setName("type_primitive_array");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual).isEqualTo("<type_primitive_array>string</type_primitive_array>");
+        }
+    }
+
+    @Nested
+    class TestObjectSchema {
+
+        @Test
+        void type_object_array() {
+            ObjectSchema itemSchema = new ObjectSchema();
+            itemSchema.addProperty("s", new StringSchema());
+            itemSchema.addProperty("b", new BooleanSchema());
+
+            ArraySchema schema = new ArraySchema();
+            schema.setItems(itemSchema);
+            schema.setName("type_object_array");
+
+            String actual = xmlSchemaWalker.fromSchema(schema, emptyMap()).trim();
+
+            assertThat(actual)
+                    .isEqualTo("<type_object_array><b>true</b><s>string</s></type_object_array>"
```

Comment:
Still struggling wether this is really correct. Maybe array schema should be transparent? At least if it contains an object schema?

Reply:
Transparent as in `<type_object_array><array><b>true</b><s>string</s></array></type_object_array>`

Reply:
No, as in `<name-of-object-schema><b>true</b><s>string</s></name-of-object-schema>`